policy/policy.go

```diff
@@ -21,90 +21,63 @@
 package policy
 
 import (
-	"bytes"
 	"errors"
-	"fmt"
 	"strings"
-	"time"
 
 	"github.com/m3db/m3metrics/generated/proto/schema"
-	"github.com/m3db/m3x/time"
 )
 
 const (
-	// InitPolicyVersion is the version of an uninitialized policy.
-	InitPolicyVersion = -1
-
-	// DefaultPolicyVersion is the version for the default policy.
-	DefaultPolicyVersion = 0
-
-	resolutionRetentionSeparator = ":"
+	policyAggregationTypeSeparator = "|"
 )
 
 var (
-	// EmptyPolicy represents an empty policy.
-	EmptyPolicy Policy
-
-	// DefaultStagedPolicies represents a default staged policies.
-	DefaultStagedPolicies StagedPolicies
-
-	// DefaultPoliciesList represents a default policies list.
-	DefaultPoliciesList = PoliciesList{DefaultStagedPolicies}
+	// DefaultPolicy represents a default policy.
+	DefaultPolicy Policy
 
 	errNilPolicySchema     = errors.New("nil policy schema")
 	errInvalidPolicyString = errors.New("invalid policy string")
 )
 
-// Policy represents the resolution and retention period metric datapoints
-// are stored at.
+// Policy contains a policy and a list of custom aggregation types.
 type Policy struct {
-	resolution Resolution
-	retention  Retention
+	sp    StoragePolicy
```

Comment:
Can we just embed the StoragePolicy here? e.g.,
```
type Policy struct {
  StoragePolicy
  AggregationID
}
```
This way you can still do Policy.Resolution() etc, which is nice